uwblueprint/starter-code-v2
# Configure SQLAlchemy migrations

## Notion ticket link

[Configure SQLAlchemy Migrations](https://www.notion.so/uwblueprintexecs/Configure-SQLAlchemy-Migrations-947d46e713ca4cd7a72e40095064609b)



## Implementation description
* Installed and initialized Flask-Migrate, created initial migration file
* Wrote documentation: https://uwblueprint.github.io/starter-code-v2/docs/migrations/python-migrations.html



## Steps to test
1. Run a `psql` shell in the DB container, drop all tables and enums in the database
```sql
-- drop commands
DROP TABLE table-name;
DROP TYPE enum-name;

-- display table, display types
\dt
\dT+
```
2. Pull from Vault (or add `FLASK_APP=app` to the root `.env` file)
3. Rebuild containers and run again if it was previously running (to pick up the new env variable). In the `py-backend` container, run the initial migration (`docker exec -it starter-code-v2_py-backend_1 /bin/bash`):
```
flask db upgrade
```
4. Check that the `entities` and `users` tables were created successfully
5. Add a model in `backend/python/app/models` called `Test`
```python
# test.py

from . import db


class Test(db.Model):
    __tablename__ = "test"

    id = db.Column(db.Integer, primary_key=True, nullable=False)
    string_field = db.Column(db.String, nullable=False)
```
6. Add `from .test import Test` to line 13 of `backend/python/app/models/__init__.py`
7. Generate the migration file, then run it. Check the database to see if the table was created successfully
```
flask db migrate -m "Create test table"
flask db upgrade
```
8. Add a new column to the `Test` model
```
int_field = db.Column(db.Integer, nullable=False)
```
9. Generate the migration file, then run it. Check the database to see if the column was added successfully
```
flask db migrate -m "Add int_field column to test"
flask db upgrade
```



## What should reviewers focus on?
Most of the files here are auto-generated, I would just focus on testing and reading the documentation.


## Checklist
